agent/reconciler/host_reconciler.go

```diff
@@ -189,9 +189,15 @@ func (r *HostReconciler) cleank8sdirectories(ctx context.Context) error {
 func (r *HostReconciler) hostCleanUp(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) error {
 	logger := ctrl.LoggerFrom(ctx)
 	logger.Info("cleaning up host")
-	err := r.resetNode(ctx, byoHost)
-	if err != nil {
-		return err
+
+	k8sComponentsInstallationSucceeded := conditions.Get(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded)
+	if k8sComponentsInstallationSucceeded != nil && k8sComponentsInstallationSucceeded.Status == corev1.ConditionTrue {
```

Comment:
I think this is good enough code to break it down into multiple functions. 
`hostCleanup` is doing 3 things:
1. execute kubeadm reset
2. uninstall k8s components
3. remove a bunch of annotations

Both 1 & 2 can be skipped if `k8sComponentsInstallationSucceeded` is not true

@dharmjit Do you see any concern with this? I don't think there is anything as partial `Install` successful. So I'm okay to skip uninstall if this condition is not true.

Reply:
@anusha94 I agree as `bootstrapK8sNode` comes after `installK8sComponents`, It is okay to skip both 1 and 2 if `k8sComponentsInstallationSucceeded` is not true.

Reply:
Have realigned `hostCleanup` to now skip both node reset and k8s component uninstallation if `k8sComponentsInstallationSucceeded` is not true